warehouse/models/intermediate/transit_database/int_transit_database__transit_data_quality_issues.sql

```diff
@@ -0,0 +1,278 @@
+        AND join_gtfs_datasets_at_resolution.issue_creation_time > dim_services._valid_from
+),
+
+join_services_at_resolution AS (
+    SELECT
+        join_services_at_creation.source_record_id,
+        service_key,
+        service_key_at_creation,
+        dim_services.key AS service_key_at_resolution,
+        dim_services.name AS service_name,
+        dim_services.source_record_id AS service_source_record_id,
+        description,
+        issue_type_key,
+        gtfs_dataset_key_at_creation,
+        gtfs_dataset_key_at_resolution,
+        status,
+        issue__,
+        resolution_date,
+        assignee,
+        issue_creation_time,
+        waiting_over_a_week_,
+        created_by,
+        qc__num_services,
+        qc__num_issue_types,
+        qc_checks,
+        waiting_on_someone_other_than_transit_data_quality_,
+        caltrans_district__from_operating_county_geographies___from_services_,
+        is_open,
+        last_update,
+        last_update_month,
+        last_update_year,
+        status_notes,
+        waiting_since,
+        outreach_status,
+        should_wait_until
+    FROM join_services_at_creation
+    LEFT JOIN dim_services
+        ON join_services_at_creation.service_key = dim_services.source_record_id
+        AND CAST(join_services_at_creation.resolution_date AS TIMESTAMP) < dim_services._valid_to
+        AND CAST(join_services_at_creation.resolution_date AS TIMESTAMP) > dim_services._valid_from
+),
+
+join_issue_types AS (
+    SELECT
+        join_services_at_resolution.source_record_id,
+        description,
+        issue_type_key,
+        distinct_issue_types.name as issue_type_name,
+        gtfs_dataset_key_at_creation,
+        gtfs_dataset_key_at_resolution,
+        status,
+        issue__,
+        service_key_at_creation,
+        service_key_at_resolution,
+        resolution_date,
+        assignee,
+        issue_creation_time,
+        waiting_over_a_week_,
+        created_by,
+        qc__num_services,
+        qc__num_issue_types,
+        qc_checks,
+        waiting_on_someone_other_than_transit_data_quality_,
+        caltrans_district__from_operating_county_geographies___from_services_,
+        is_open,
+        last_update,
+        last_update_month,
+        last_update_year,
+        status_notes,
+        waiting_since,
+        outreach_status,
+        should_wait_until
+    FROM join_services_at_resolution
+    INNER JOIN (select distinct source_record_id, name, _is_current, _valid_from, _valid_to from dim_issue_types) as distinct_issue_types
+        ON join_services_at_resolution.issue_type_key = distinct_issue_types.source_record_id
+),
+
+int_transit_database__transit_data_quality_issues AS (
+    SELECT
+        {{ dbt_utils.generate_surrogate_key(['source_record_id', 'gtfs_dataset_key_at_creation', 'gtfs_dataset_key_at_resolution', 'service_key_at_creation', 'service_key_at_resolution']) }} AS key,
```

Comment:
I think that it would actually be safer to use `gtfs_dataset.source_record_id` and `service.source_record_id` in the composite key (without the `_key`s) just to confirm that you're only getting one combination here (make sure you didn't accidentally get any fanout in the other joins.... by testing on the versioned keys there's more room to end up with something unexpected)